docs/design_doc_blueprint.md

```diff
@@ -0,0 +1,44 @@
+# Title
+
+> A cool name for your Project
+
+Author: [Your-Name](mailto:your-name@reachable.com)
+
+> If there are multiple authors, just list them all
+
+## Summary
+
+A short summary of your project/re-design/component and what problems it 
+addresses in about 3-10 sentences.
+
+## State of the Feature as of `$VERSION` (optional)
+
+The state of the feature you want to improve and where it currently falls 
+short. If there's nothing to compare to, leave it out.
+
+## Goals and non-goals
+
+Direct goals of the project and features deemed not worth pursuing.
+
+## Overview
```

Comment:
Might also be useful to add the following:
* What assumed knowledge is there
  * Is there any place you can read up on any assumed knowledge that might be required to fully understand this document?
* Related work
  * Eg. Have any other VCS's addressed this problem? If so:
    * How did they do it
    * Pros / cons of the way they did it, would it work for jj?

Reply:
I added a prior and related work section.